contributors/design-proposals/daemonset-update.md

```diff
@@ -129,18 +127,35 @@ type DaemonSetStatus struct {
 
 	// UpdatedNumberScheduled is the total number of nodes that are running updated
 	// daemon pod
+	// +optional
 	UpdatedNumberScheduled int32 `json:"updatedNumberScheduled"`
 
 	// NumberAvailable is the number of nodes that should be running the
 	// daemon pod and have one or more of the daemon pod running and
 	// available (ready for at least minReadySeconds)
+	// +optional
 	NumberAvailable int32 `json:"numberAvailable"`
 
 	// NumberUnavailable is the number of nodes that should be running the
 	// daemon pod and have non of the daemon pod running and available
 	// (ready for at least minReadySeconds)
+	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
 }
+
+type DaemonSet struct {
```

Comment:
The reason that I put it under `DaemonSet` instead of `DaemonSetSpec` is that `TemplateGeneration` is read-only and generated by system - wouldn't it be a bit odd that such field exists in the spec, where users will specify expected state there?